### PR TITLE
Refactoring and maintenance for new drivers

### PR DESCRIPTION
**Pull request by rexut**
This is more an improvements for last PR #248 and #249 then new feature set.

### CODE REVIEW

File: drivers/i2c/i2c_sc18is604.c [hunk 1]
```diff
@@ -385,20 +382,20 @@ static int i2c_sc18is604_init(const struct device *dev)
 	k_poll_signal_init(&data->interrupt_signal);
 
 	/* Register interrupt callback with parent MFD */
-	gpio_init_callback(&data->interrupt_callback,
-					   i2c_sc18is604_interrupt_callback,
-					   0xffffff);
-	ret = mfd_sc18is604_add_callback(config->parent_dev, &data->interrupt_callback);
+	gpio_init_callback(&data->interrupt_cb,
+			   i2c_sc18is604_interrupt_callback, 0xffffff);
```
Comment (Irockasingranite) [inline drivers/i2c/i2c_sc18is604.c:386]: This mask should probably be the full 32 bits:
```suggestion
	gpio_init_callback(&data->interrupt_cb,
			   i2c_sc18is604_interrupt_callback, 0xffffffff);
```

**Review by Irockasingranite — CHANGES_REQUESTED** [review]
Just a couple of things I found while testing these drivers. The change to treating a missing reset pin on an SC18IS604 is slightly different to the one I made locally, but it looks good.

File: drivers/mfd/mfd_sc18is604.c
[the remark this answers is not in the document]
Comment (rexut) [reply drivers/mfd/mfd_sc18is604.c:1]: done

File: drivers/mfd/mfd_sc18is604.c [hunk 2]
```diff
@@ -178,7 +211,7 @@ static void mfd_sc18is604_interrupt_callback(const struct device *dev,
 static int mfd_sc18is604_check_chipid(const struct device *dev,
```
[the remark this answers is not in the document]
Comment (rexut) [reply drivers/mfd/mfd_sc18is604.c:212]: I've add the `mfd_sc18is604_clear_interrupt_source()` call inside the "Software Reset" branch in `mfd_sc18is604_init()`. @Irockasingranite feel free to mark the conversation as resolved, when this looks good enough to you.

File: drivers/i2c/i2c_sc18is604.c (hunk 1 above)
Comment (rexut) [reply drivers/i2c/i2c_sc18is604.c]: done

File: drivers/mfd/mfd_sc18is604.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (Irockasingranite) [reply drivers/mfd/mfd_sc18is604.c:212]: Yes, that should do the trick.

**Review by Irockasingranite — APPROVED** [review]

**Pull request merged**